app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/masterpasswordgenerator/MasterPasswordGeneratorScreen.kt

```diff
@@ -0,0 +1,231 @@
+        },
+    ) { innerPadding ->
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .verticalScroll(rememberScrollState())
+                .padding(innerPadding),
+        ) {
+            MasterPasswordGeneratorContent(
+                generatedPassword = state.generatedPassword,
+                onGenerateNewPassword = remember(viewModel) {
+                    {
+                        viewModel.trySendAction(
+                            MasterPasswordGeneratorAction.GeneratePasswordClickAction,
+                        )
+                    }
+                },
+                onLearnToPreventLockout = remember(viewModel) {
+                    {
+                        viewModel.trySendAction(
+                            MasterPasswordGeneratorAction.PreventLockoutClickAction,
+                        )
+                    }
+                },
+                modifier = Modifier.padding(horizontal = 16.dp),
+            )
+            Spacer(modifier = Modifier.navigationBarsPadding())
+        }
+    }
+}
+
+@Composable
+private fun MasterPasswordGeneratorContent(
+    generatedPassword: String,
+    onGenerateNewPassword: () -> Unit,
+    onLearnToPreventLockout: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier.fillMaxWidth(),
+    ) {
+        Text(
+            text = stringResource(R.string.generate_master_password),
+            style = MaterialTheme.typography.headlineSmall,
+            color = MaterialTheme.colorScheme.onSurface,
+        )
+        Spacer(modifier = Modifier.height(24.dp))
+        BitwardenTextField(
+            label = "",
+            value = generatedPassword,
+            onValueChange = {},
+            readOnly = true,
+            shouldAddCustomLineBreaks = true,
+            textStyle = LocalNonMaterialTypography.current.sensitiveInfoSmall,
+            visualTransformation = nonLetterColorVisualTransformation(),
+            modifier = Modifier.fillMaxWidth(),
+        )
+
+        Spacer(modifier = Modifier.height(12.dp))
+        BitwardenFilledButtonWithIcon(
+            label = stringResource(R.string.generate_button_label),
+            onClick = onGenerateNewPassword,
+            icon = rememberVectorPainter(id = R.drawable.ic_generator),
+            modifier = Modifier.fillMaxWidth(),
+        )
+        Spacer(modifier = Modifier.height(12.dp))
+        Text(
+            text = stringResource(R.string.write_this_password_down_and_keep_it_somewhere_safe),
+            style = MaterialTheme.typography.bodySmall,
+            color = MaterialTheme.colorScheme.onSurfaceVariant,
+        )
+        Text(
+            text = stringResource(R.string.learn_about_other_ways_to_prevent_account_lockout),
+            style = MaterialTheme.typography.labelMedium,
+            color = MaterialTheme.colorScheme.primary,
+            modifier = Modifier.clickable(onClick = onLearnToPreventLockout),
```

Comment:
This should be a `BitwardenClickableText` to ensure the ripple is present.

You can see an example in the `CreateAccountScreen`